src/test/scala/uk/gov/hmrc/perftests/disareturns/ReportingWindowRequests.scala

```diff
@@ -16,22 +16,28 @@
 
 package uk.gov.hmrc.perftests.disareturns
 
-import io.gatling.core.Predef._
-import io.gatling.http.Predef._
-import io.gatling.http.request.builder.HttpRequestBuilder
 import play.api.libs.json.{JsObject, Json}
-import uk.gov.hmrc.performance.conf.ServicesConfiguration
+import play.api.libs.ws.DefaultBodyWritables.writeableOf_String
+import play.api.libs.ws.ahc.StandaloneAhcWSClient
 import uk.gov.hmrc.perftests.disareturns.constant.AppConfig._
 import uk.gov.hmrc.perftests.disareturns.constant.Headers.reportingWindowHeaders
 
-object ReportingWindowRequests extends ServicesConfiguration {
+import javax.inject.Singleton
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class ReportingWindowRequests(ws: StandaloneAhcWSClient)(implicit ec: ExecutionContext) {
   val reportingWindowPayload: JsObject = Json.obj("reportingWindowOpen" -> true)
 
-  val setReportingWindowsOpen: HttpRequestBuilder =
-    http("Set reporting window as Open")
-      .post(s"$disaReturnsStubHost$reportingWindowPath")
-      .headers(reportingWindowHeaders)
-      .body(StringBody(reportingWindowPayload.toString()))
-      .check(status.is(204))
-      .silent
+  def setReportingWindowsOpen(): Future[Unit] = {
+    val url = s"$disaReturnsStubHost$reportingWindowPath"
+    ws
+      .url(url)
+      .addHttpHeaders(reportingWindowHeaders.toSeq: _*)
+      .post(reportingWindowPayload.toString())
+      .map { response =>
+        if (response.status != 204) Right(())
+        else Left(s"Failed to set the reporting window open. Status ${response.status}, body: ${response.body}")
```

Comment:
Return type here is a mismatch

The .map { ... } block returns Right(()) or Left(String), which is of type Either[String, Unit].

But the method signature declares Future[Unit]?

I think Scala will implicitly discard the Either value and just return Future[Unit], meaning the failure info is lost.